kite-data/kite-data-core/src/main/java/org/kitesdk/data/DatasetException.java

```diff
@@ -46,4 +46,11 @@ public DatasetException(Throwable t) {
     super(t);
   }
 
+  protected static String format(String message, Object... args) {
+    String[] argStrings = new String[args.length];
+    for (int i = 0; i < args.length; i += 1) {
+      argStrings[i] = String.valueOf(args[i]);
+    }
+    return String.format(String.valueOf(message), (Object[]) argStrings);
```

Comment:
What does this provide over `String.format(message, args)`? Message shouldn't be null. Also, if any of the args are null then they can be handled by the conversion specified in http://docs.oracle.com/javase/7/docs/api/java/util/Formatter.html#syntax